webapp/src/store/currentWorkspace.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSlice, createAsyncThunk, PayloadAction} from '@reduxjs/toolkit'
+
+import {default as client} from '../octoClient'
+import {IWorkspace} from '../blocks/workspace'
+
+import {RootState} from './index'
+
+export const fetchCurrentWorkspace = createAsyncThunk(
+    'currentWorkspace/fetch',
+    async () => client.getWorkspace(),
+)
+
+const currentWorkspaceSlice = createSlice({
+    name: 'currentWorkspace',
+    initialState: {value: null} as {value: IWorkspace|null},
+    reducers: {
+        setWorkspace: (state, action: PayloadAction<IWorkspace>) => {
+            state.value = action.payload
+        },
+    },
+    extraReducers: (builder) => {
+        builder.addCase(fetchCurrentWorkspace.fulfilled, (state, action) => {
```

Comment:
Why not typing the action here? (same in the other `extraReducers`)

Reply:
What do you mean?